nuxeo-core/nuxeo-core-storage-sql/nuxeo-core-storage-sql/src/main/java/org/nuxeo/ecm/core/storage/sql/PersistenceContext.java

```diff
@@ -1543,11 +1544,13 @@ public Serializable checkIn(Node node, String label, String checkinComment) {
          * Do the copy without non-complex children, with null parent.
          */
         Serializable id = node.getId();
-        CopyResult res = mapper.copy(new IdWithTypes(node), null, null, null);
+        // complex (special) children are snapshot when checking in
+        boolean excludeSpecialChildren = false;
+        CopyResult res = mapper.copy(new IdWithTypes(node), null, null, null, excludeSpecialChildren);
         Serializable newId = res.copyId;
         markInvalidated(res.invalidations);
         // add version as a new child of its parent
-        SimpleFragment verHier = getHier(newId, false);
+        SimpleFragment verHier = getHier(newId, excludeSpecialChildren);
```

Comment:
This is wrong, the second argument of `getHier` is `allowAbsent` and shouldn't be changed here.